src/editor/Collaboration.js

```diff
@@ -160,18 +166,136 @@ define(function (require, exports, module) {
+     *
+     * @param {!string} path
+     * @return {$.Promise} A promise object that will be resolved when the buffer array for the file
+     * is cleared or rejected with a FileSystemError if the file is not yet open and can't be read from disk.
+     */
+    function clearFile(path) {
+        var result = new $.Deferred();
+        var file = FileSystem.getFileForPath(path);
+        if(!_webrtc || !_buffer || !_buffer[path] || _buffer[path].length === 0) {
+            FileUtils.readAsText(file)
+                .done(function (rawText, readTimestamp) {
+                    result.resolve(rawText, readTimestamp);    
+                })
+                .fail(function (fileError) {
+                    result.reject(fileError);
+                });
+            return result.promise();
+        }
+        FileUtils.readAsText(file)
+            .done(function (text, readTimestamp) {
+                var numberOfChanges = 0;
+                _buffer[path].forEach(function(delta) {
+                    numberOfChanges++;
+                    console.log("added change for file");
+                    var start = _indexFromPos(delta.from, text);
+                    // apply the delete operation first
+                    if (delta.removed.length > 0) {
+                        var delLength = 0;
+                        for (var i = 0; i < delta.removed.length; i++) {
+                         delLength += delta.removed[i].length;
+                        }
+                        delLength += delta.removed.length - 1;
+                        var from = _posFromIndex(start, text);
+                        var to = _posFromIndex(start + delLength, text);
+                        text = _replaceRange('', from, to, text);
+                    }
+                    // apply insert operation
+                    var param = delta.text.join('\n');
+                    var from = _posFromIndex(start, text);
+                    var to = from;
+                    text = _replaceRange(param, from, to, text);
+                });
+                FileUtils.writeText(file, text)
+                    .done(function() {
+                        console.log("writting to file which is not open in editor for path " + path);
+                        _buffer[path].splice(0, numberOfChanges);
+                        result.resolve(text, readTimestamp);
+                    })
+                    .fail(function(err) {
+                        result.reject(err);
+                    });
+            })
+            .fail(function (fileError) {
+                result.reject(fileError);
+            });
+        return result.promise();
+    }
+
     function _getOpenCodemirrorInstance(fullPath) {
-        var masterEditor = EditorManager.getCurrentFullEditor();
-        if(masterEditor.getFile().fullPath === fullPath) {
-            return masterEditor._codeMirror;
+        var doc = DocumentManager.getOpenDocumentForPath(fullPath);
```

Comment:
Shifted to syncing all open codemirror instances instantly rather than just the one open because the control doesn't reach [here](https://github.com/mozilla/brackets/blob/collaboration/src/document/DocumentManager.js#L337) for these files.
If a file has a codemirror instance attached to it, the control reaches [here](https://github.com/mozilla/brackets/blob/collaboration/src/document/DocumentManager.js#L333) and it doesn't read from the filesystem.